### ISSUE TITLE
Prefer dict literal over `dict` constructor

### ISSUE THREAD

**Issue by hofrob** [PyCQA/pylint#7690]
### Current problem

Constructing `dict`s using the constructor should be avoided. It has no advantages and may be a bit slower.

## Invalid code

```py
# construct dict using keywords
dict(foo="bar")
# construct dict using a dict
dict(**boo)
```

## Valid code 

```py
# dict literal
{"foo": "bar"}

# casting values to a dict
dict(some_method())

# methods that are not the builtin `dict`
def dict(baz):
    ...

dict(1)
```

### Desired solution

Create a message that finds and highlights invalid code from above. Maybe as an extension of `R1735 use-dict-literal`.

**Comment by nickdrozd** [comment]
I just came across this new check (from running latest in Astroid). It seems awfully opinionated to me. Is it the official stance of Pylint that this syntax is bad? Is it old or deprecated or something? The [official documentation](https://docs.python.org/3/tutorial/datastructures.html#dictionaries) doesn't say much about it, just that "when the keys are simple strings, it is sometimes easier to specify pairs using keyword arguments". So one reason you might want it is if you don't want extra strings floating around.

It's certainly not a widely used way to make dictionaries, but I don't see why it should be recommended against. Is it really slower?

**Comment by hofrob** [comment]
https://github.com/PyCQA/pylint/blob/main/doc/data/messages/u/use-dict-literal/details.rst

If this was a feature where the devs decided that one version looks better than the other and style was the only consideration, I'd agree. But this was not the case here at all.

Literals are faster. And faster is better when there's so little difference in style (my personal opinion). It would even make sense to expand this rule to sets and lists (maybe this is already working or in the works).

**Comment by nickdrozd** [comment]
Okay, I'm agreed then on the other constructors as well.
